spec/lib/recog/nizer_spec.rb

```diff
@@ -36,6 +46,84 @@
     end
   end
 
+  describe ".match_all_db" do
+    File.readlines(File.expand_path(File.join('spec', 'data', 'smb_native_os.txt'))).each do |line|
+      data = line.strip
+      context "with smb_native_os:#{data}" do
+        let(:match_all_result) { subject.match_all_db(data, VALID_FILTER) }
+
+        it "returns an array" do
+          expect(match_all_result.class).to eq(::Array)
+        end
+
+        it "returns a successful match" do
+          expect(match_all_result[0]['matched'].to_s).to match(/^[A-Z]/)
```

Comment:
Here and several other places you are calling `to_s` on the `matched` value, but not for other values.  Why is this necessary?  These should already be strings and not, say, XML attributes.